audioswitch/src/main/java/com/twilio/audioswitch/bluetooth/BluetoothHeadsetManager.kt

```diff
@@ -302,11 +314,91 @@ internal constructor(
         override fun scoAction() {
             logger.d(TAG, "Attempting to disable bluetooth SCO")
             audioDeviceManager.enableBluetoothSco(false)
-            headsetState = Connected
+            headsetState = HeadsetState.Connected
         }
 
         override fun scoTimeOutAction() {
-            headsetState = AudioActivationError
+            headsetState = HeadsetState.AudioActivationError
         }
     }
 }
+
+internal class BluetoothHeadsetManager_v31plus
+
+internal constructor(
+    context: Context,
+    logger: Logger,
+    bluetoothAdapter: BluetoothAdapter,
+    audioDeviceManager: AudioDeviceManager,
+    headsetListener: BluetoothHeadsetConnectionListener? = null,
+    bluetoothScoHandler: Handler = Handler(Looper.getMainLooper()),
+    systemClockWrapper: SystemClockWrapper = SystemClockWrapper(),
+    bluetoothIntentProcessor: BluetoothIntentProcessor = BluetoothIntentProcessorImpl(),
+    headsetProxy: BluetoothHeadset? = null
+) : BluetoothHeadsetManagerDefault(
+    context,
+    logger,
+    bluetoothAdapter,
+    audioDeviceManager,
+    headsetListener,
+    bluetoothScoHandler,
+    systemClockWrapper,
+    bluetoothIntentProcessor,
+    headsetProxy
+) {
+    private val ERROR_MSG = "Bluetooth unsupported, permissions not granted"
```

Comment:
```suggestion
    private val permissionErrorMessage = "Bluetooth permissions not granted"
```

However, if you want to use all caps case you can make this a const and move to the top level file

```
private const val PERMISSION_ERROR_MESSAGE = "Bluetooth permissions not granted"

internal class BluetoothHeadsetManager_v31plus

// ...
```